src/neptune/api/operations.py

```diff
@@ -0,0 +1,200 @@
+#
+# Copyright (c) 2024, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+__all__ = [
+    "ProtoSerializable",
+    "Run",
+    "FloatValue",
+    "LogFloats",
+    "AssignInteger",
+    "AssignFloat",
+    "AssignBool",
+    "AssignString",
+    "AssignDatetime",
+    "RunOperation",
+]
+
+
+import abc
+from dataclasses import dataclass
+from datetime import datetime
+from typing import (
+    List,
+    Optional,
+)
+
+from google.protobuf import timestamp_pb2
+
+import neptune.api.proto.neptune_pb.ingest.v1.common_pb2 as common_pb2
+import neptune.api.proto.neptune_pb.ingest.v1.pub.ingest_pb2 as ingest_pb2
+
+
+class ProtoSerializable(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def create(self) -> Optional[common_pb2.Run]: ...
+
+    @property
+    @abc.abstractmethod
+    def update(self) -> Optional[common_pb2.UpdateRunSnapshot]: ...
+
+
+class SerializableCreation(ProtoSerializable, abc.ABC):
+    @property
+    def update(self) -> Optional[common_pb2.UpdateRunSnapshot]:
+        return None
+
+
+class SerializableUpdate(ProtoSerializable, abc.ABC):
+    @property
+    def create(self) -> Optional[common_pb2.Run]:
+        return None
```

Comment:
do we actually need those abstractions?
I feel like it introduces kinda loose typing (it requires to return Optional type) and as we will be introducing new messages we will have to add new classes of type `Serializable<NewMessage>` while old classes like `SerializabeCreation` will have to dummy implement new method. Also `RunOperation` doesn't use this abstraction at all.